Userland/Libraries/LibAudio/Buffer.cpp

```diff
@@ -166,4 +166,25 @@ RefPtr<Buffer> Buffer::from_pcm_stream(InputMemoryStream& stream, ResampleHelper
     return Buffer::create_with_samples(move(fdata));
 }
 
+Vector<i32> ResampleHelper::resample(Vector<i32> to_resample)
+{
+    double sample = 0;
+    Vector<i32> resampled;
+    auto to_resample_it = to_resample.begin();
+    resampled.ensure_capacity(to_resample.size() * m_ratio);
+    for (;;) {
+        while (read_sample(sample, sample)) {
+            resampled.append(static_cast<i32>(sample));
+        }
+        ++to_resample_it;
+        if (to_resample_it.is_end())
+            break;
+        sample = *to_resample_it;
+
+        process_sample(sample, sample);
+    }
```

Comment:
I'm somewhat confused here, if there's a known number of things to resample, why is this an infinite loop?
How about this? 
```suggestion
    resampled.ensure_capacity(to_resample.size() * m_ratio);
    for (auto& sample : to_resample) {
        while (read_sample(sample, sample))
            resampled.unchecked_append(static_cast<i32>(sample));
        process_sample(sample, sample);
    }
```

Reply:
Thanks for finding that bug! I vaguely remember copying that code from somewhere else, but oh well...